pkg/resource/deploy/step.go

```diff
@@ -252,6 +252,11 @@ func (s *CreateStep) Apply(preview bool) (resource.Status, StepCompleteFunc, err
 		s.new.Outputs = outs
 	}
 
+	// If we've created a new resource as a replacement for an old one check that the ID is a new ID.
+	if s.replacing && s.new.Custom && !preview && s.new.ID == s.old.ID {
```

Comment:
Curious does it need !preview.

Also curious below `s.replacing && s.pendingDelete` check. Will it not try to delete unless `s.old.Delete = true` runs? Then maybe we should restrict the error further to `s.pendingDelete` as well as `s.replacing`?

Reply:
>Curious does it need !preview.
Probably not we won't have an ID at preview time.

> Will it not try to delete unless s.old.Delete = true runs
Correct. This ensure that the resource doesn't get deleted in the case where ID comes back the same, we error out of the engine first.

> Then maybe we should restrict the error further to s.pendingDelete as well as s.replacing?
Ah yes we probably should. If this is a CreateReplecement but "old" has already been deleted than s.pendingDelete will be false.